CreeDictionary/API/models.py

```diff
@@ -127,8 +132,15 @@ class SearchResult:
     matched_by: Language
 
     # The matched lemma
+    # todo: pass in dumb object instead to help serialization and avoid unnecessary database access
+    #   spread the fields of lemma_wordform here, and refactor word-entries.html
     lemma_wordform: "Wordform"
 
+    # looks like "/words/nipaw" "/words/nipâw?pos=xx" "/words/nipâw?full_lc=xx" "/words/nipâw?analysis=xx" "/words/nipâw?id=xx"
+    # it's the least strict url that guarantees unique match in the database
+    # see https://github.com/UAlbertaALTLab/cree-intelligent-dictionary/issues/143
+    lemma_url: str
```

Comment:
I could have avoided adding this fields but use {% url lemma_wordform %} in the template (Protip: this will actually call the `get_absolute_url` method of the instance). But I'm thinking of the "dumb object" convention.

Reply:
This adds more [accidental state](https://en.wikipedia.org/wiki/No_Silver_Bullet#Summary) to the databased, and I'm pretty wary of that. See my comment about dumb and smart objects.